### PR TITLE
solution

### PR DESCRIPTION
**Pull request by uzlss**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,25 +1,74 @@
...
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Callable, name: str) -> None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: owner is type
instance is object

File: app/main.py [hunk 2]
```diff
@@ -1,25 +1,74 @@
...
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Callable, owner: Callable) -> Any:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Callable, value: Any) -> None:
+        if value in range(self.min_amount, self.max_amount + 1):
+            setattr(instance, self.protected_name, value)
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: otherwise you need to raise an Exception

File: app/main.py [hunk 3]
```diff
@@ -1,25 +1,74 @@
...
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: it accepts type of SlideLimitationValidator, not instance

File: app/main.py [hunk 4]
```diff
@@ -1,25 +1,74 @@
...
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, instance: Visitor) -> bool:
+        return len(
+            self.limitation_class(
+                instance.age,
+                instance.weight,
+                instance.height
+            ).__dict__) == 3
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: use try-except after implement raise exception inside `__set__`

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 4 above)
Comment (uzlss) [reply app/main.py]: flake8 says that I should not
I would rather use "finally" cuz there will be "return" anyway

Comment (uzlss) [reply app/main.py]: no, it does not work :(

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**